src/NuGet.Core/NuGet.Protocol/Plugins/InboundRequestContext.cs

```diff
@@ -222,12 +170,35 @@ public void BeginFaultAsync(Message request, Exception exception)
                             responseHandler,
                             _cancellationToken);
                     }
+                    catch (OperationCanceledException) when (_cancellationToken.IsCancellationRequested)
+                    {
+                        var response = MessageUtilities.Create(request.RequestId, MessageType.Cancel, request.Method);
+
+                        await _connection.SendAsync(response, CancellationToken.None);
+                    }
                     catch (Exception ex)
                     {
                         BeginFaultAsync(request, ex);
                     }
                 },
                 _cancellationToken);
         }
+
+        /// <summary>
+        /// Cancels an inbound request.
+        /// </summary>
+        public void Cancel()
+        {
+            try
+            {
+                _cancellationTokenSource.Cancel();
+            }
+            catch (AggregateException)
```

Comment:
Per MSDN, `Cancel()` can throw an `AggregateException`.  The intent behind this is to make this "reasonably" exception-safe.  (For example, don't catch `AccessViolationException`.)  This code is called by a thread which reads the plugin's standard output stream and completes outstanding tasks.  Should an exception propagate up, it would exit this thread, resulting in request timeouts and an overall operation failure.